apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/RequestHeaders.java

```diff
@@ -0,0 +1,207 @@
+package gov.cms.bfd.server.war.commons;
+
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.server.exceptions.InvalidRequestException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * A per request instance holds all resource (FHIR) request headers, such as: "includeIdentifiers"
+ * {@link PatientResourceProvider#HEADER_NAME_INCLUDE_IDENTIFIERS} "includeAddressFields" {@link
+ * PatientResourceProvider#HEADER_NAME_INCLUDE_ADDRESS_FIELDS} which serve as part of BFD API
+ */
+public class RequestHeaders {
+  RequestDetails requestDetails;
+  Map<String, Object> headerNVs = new HashMap<String, Object>();
+
+  private RequestHeaders(RequestDetails requestDetails) {
+    this.requestDetails = requestDetails;
+    // parse headers
+    CommonHeaders.FHIR_REQUEST_HEADERS.forEach(
+        (h) -> {
+          String v = this.requestDetails.getHeader(h);
+          if (h.equals(CommonHeaders.HEADER_NAME_INCLUDE_ADDRESS_FIELDS)) {
+            this.headerNVs.put(h, returnIncludeAddressFieldsValue(v));
+          }
+          if (h.equals(CommonHeaders.HEADER_NAME_INCLUDE_IDENTIFIERS)) {
+            this.headerNVs.put(h, returnIncludeIdentifiersValues(v));
+          }
+        });
+  }
+
+  /**
+   * instantiate an empty RH, used by tests
+   *
+   * @param requestDetails
+   */
+  private RequestHeaders() {}
+
+  /**
+   * currently used by internal code to create a RH instance with ad hoc header/value pairs for
+   * testing code
+   *
+   * @param hdrValues the header/value pairs
+   */
+  private RequestHeaders(String... hdrValues) {
+    // convert h1 v1, h2 v2, ... into nv pairs
+    if (hdrValues == null || hdrValues.length % 2 != 0) {
```

Comment:
@jfuqian Can you detail the reason for failing requests if these request headers arent populated? These should be optional, and the default to false I believe. What was your thinking with this?

Reply:
that constructor is used for tests only.